salt/_states/kubeconfig.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+
+from base64 import b64encode
+import os
+import stat
+import yaml
+
+__virtualname__ = 'metalk8s_kubeconfig'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def _validateKubeConfig(filename, expected_ca_data, expected_api_server):
+    """Validate a kubeconfig filename.
+
+    Validate that the kubeconfig provided by filename
+    is conform with config.
+
+    This function is used for managed idempotency.
+
+    :return: True if the kubeconfig file matches expectation
+             False otherwise (ie need to be regenerated)
+    """
+    # Verify that the file does not exist
+    if not os.path.isfile(filename):
+        return False
+
+    # Verify that the mode is 600
+    if stat.S_IMODE(os.stat(filename).st_mode) != 0o600:
+        return False
+
+    try:
+        with open(filename, 'r') as fd:
+            kubeconfig = yaml.safe_load(fd)
+    except Exception:
+        return False
+
+    # Verify that the current CA cert on disk matches the expected CA cert
+    # and the API Server on the existing file match with the expected
+    try:
+        cluster_info = kubeconfig['clusters'][0]['cluster']
+        current_ca_data = cluster_info['certificate-authority-data']
+        current_api_server = cluster_info['server']
+    except (KeyError, IndexError):
+        return False
+
+    if current_ca_data != expected_ca_data:
+        return False
+
+    if current_api_server != expected_api_server:
+        return False
+
+    return True
+
+
+def managed(name,
+            ca_server,
+            signing_policy,
+            client_cert_info,
+            apiserver,
+            cluster,
+            **kwargs):
+    """Generate kubeconfig file with identities for control plane components"""
+    ret = {
+        'name': name,
+        'changes': {},
+        'comment': "",
+        'result': True,
+    }
+
+    # Get the CA cert from mine
+    try:
+        b64_ca_cert = __salt__['mine.get'](
```

Comment:
After checking salt command I suggest to use `x509.get_signing_policy` to get the CA certificate I think it's better

Reply:
Yep, you're right it's better, I will change it and I don't even need to store the certif in the mine

Reply:
Seen together but does not work actually as `x509.get_signing_policy` return only the result on the salt master but not the minion, so we can't query the ca cert from minion with `x509.get_signing_policy`